File: JavaScript/795-Number of Subarrays with Bounded Maximum.js
/**
 * @param {number[]} A
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var numSubarrayBoundedMax = function(A, L, R) {
    
    let findSub = (bound) =>{
        let curr = 0, res = 0;
        for (let val of A){
            if (val <= bound){
                curr += 1;
                res += curr;
            }else{
                curr = 0;
            }
        }
        return res;
    }
    return findSub(R) - findSub(L-1);
};

File: 795-Number of Subarrays with Bounded Maximum.py
class Solution(object):
    def numSubarrayBoundedMax(self, A, L, R):
        """
        :type A: List[int]
        :type L: int
        :type R: int
        :rtype: int
        """
# Time complexity:O(n), Space complexity:O(1)        
# findSub(B) is the number of subarrays that have all elements less than or equal to B

        def findSub(bound):
            res = curr = 0
            for val in A:
                if val <= bound:
                    curr += 1
                else:
                    curr = 0
                res += curr
            return res
        
        return findSub(R) - findSub(L-1)